datafusion/core/src/catalog/information_schema.rs

```diff
@@ -579,3 +613,45 @@ impl From<InformationSchemaColumnsBuilder> for MemTable {
         MemTable::try_new(schema, vec![vec![batch]]).unwrap()
     }
 }
+
+struct InformationSchemaSettingsBuilder {
+    names: StringBuilder,
+    settings: StringBuilder,
+}
+
+impl InformationSchemaSettingsBuilder {
+    fn new() -> Self {
+        Self {
+            names: StringBuilder::new(),
+            settings: StringBuilder::new(),
+        }
+    }
+
+    fn add_setting(&mut self, name: impl AsRef<str>, setting: impl AsRef<str>) {
+        self.names.append_value(name.as_ref());
+        self.settings.append_value(setting.as_ref());
+    }
+}
+
+impl From<InformationSchemaSettingsBuilder> for MemTable {
+    fn from(value: InformationSchemaSettingsBuilder) -> MemTable {
+        let schema = Schema::new(vec![
+            Field::new("name", DataType::Utf8, false),
+            Field::new("setting", DataType::Utf8, false),
+        ]);
+
+        let InformationSchemaSettingsBuilder {
+            mut names,
+            mut settings,
+        } = value;
+
+        let schema = Arc::new(schema);
+        let batch = RecordBatch::try_new(
+            schema.clone(),
+            vec![Arc::new(names.finish()), Arc::new(settings.finish())],
+        )
+        .unwrap();
+
+        MemTable::try_new(schema, vec![vec![batch]]).unwrap()
+    }
+}
```

Comment:
implement `InformationSchemaSettingsBuilder`
only two columns, `name` and `setting` supported now

this is what postgresql has
```bash
show all;

                  name                  |                                 setting                                  |                                                               description                                                               
----------------------------------------+--------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------
 allow_system_table_mods                | off                                                                      | Allows modifications of the structure of system tables.
 application_name                       | psql                                                                     | Sets the application name to be reported in statistics and logs.
 archive_cleanup_command                |                                                                          | Sets the shell command that will be executed at every restart point.
 archive_command                        | (disabled)                                                               | Sets the shell command that will be called to archive a WAL file.
 archive_mode                           | off                                                                      | Allows archiving of WAL files using archive_command.
 archive_timeout                        | 0                                                                        | Forces a switch 
```